bob-apps/bin/bobp.rs

```diff
@@ -438,46 +535,73 @@ fn print_periodic_stat(
-        let get_error = stat.get_error_count.load(Ordering::Relaxed);
-        let put_spd = put_count_spd as f64 * k;
-        let get_spd = get_size.get_diff() as f64 / 1024.0;
-        if put_spd > 0.0 {
+        if print_put {
+            let d_put = put_count.get_diff();
+            let put_count_spd = d_put * 1000 / period_ms;
+            let cur_st_put_time = put_time_st.get_diff() as f64;
+            let cur_st_put_count = put_count_st.get_diff() as f64;
+            let put_error = stat.put_error_count.load(Ordering::Relaxed);
+            let put_spd = put_count_spd as f64 * k;
+            println!("put:   {:>6} rps | err {:5} | {:>6.2} kb/s | lat {:>6.2} ms", 
+                put_count_spd,
+                put_error,
+                put_spd,
+                finite_or_default(cur_st_put_time / cur_st_put_count / 1e9));
+
             put_speed_values.push(put_spd);
         }
-        if get_spd > 0.0 {
+        if print_get {
+            let d_get = get_count.get_diff();
+            let get_count_spd = d_get * 1000 / period_ms;
+            let cur_st_get_time = get_time_st.get_diff() as f64;
+            let cur_st_get_count = get_count_st.get_diff() as f64;
+            let get_error = stat.get_error_count.load(Ordering::Relaxed);
+            let get_spd = get_size.get_diff() as f64 / 1024. / sec;
+            println!("get:   {:>6} rps | err {:5} | {:>6.2} kb/s | lat {:>6.2} ms", 
+                get_count_spd,
+                get_error,
+                get_spd,
+                finite_or_default(cur_st_get_time / cur_st_get_count / 1e9));
+
             get_speed_values.push(get_spd);
         }
-        println!(
-            "put: {:>6} rps  | get {:>6} rps   | put err: {:5}     | get err: {:5}\r\n\
-            put: {:>6.2} kb/s | get: {:>6.2} kb/s | put lat: {:>6.2} ms | get lat: {:>6.2} ms",
-            put_count_spd,
-            get_count_spd,
-            put_error,
-            get_error,
-            put_spd,
-            get_spd,
-            finite_or_default(cur_st_put_time / cur_st_put_count / 1e9),
-            finite_or_default(cur_st_get_time / cur_st_get_count / 1e9)
-        );
+        if print_exist {
+            let d_exist = exist_count.get_diff();
+            let exist_count_spd = d_exist * 1000 / period_ms;
+            let cur_st_exist_time = exist_time_st.get_diff() as f64;
+            let cur_st_exist_count = exist_count_st.get_diff() as f64;
+            let exist_error = stat.get_error_count.load(Ordering::Relaxed);
+            let exist_spd = exist_size.get_diff() as f64 / 1024.0 / sec;
+            println!("exist: {:>6} rps | err {:5} | {:>6.2} kb/s | lat {:>6.2} ms", 
+                exist_count_spd,
+                exist_error,
+                exist_spd,
+                finite_or_default(cur_st_exist_time / cur_st_exist_count / 1e9));
+
+            exist_speed_values.push(exist_spd);
+        }
+        println!("-------------");
```

Comment:
This separator is redundant. Especially if all output information is placed in one line

Reply:
In the current state output isn't placed in one line, but exist, get, put are printed in separate lines

Reply:
In practice, only `put`, `get` and `exist` modes are used. 
`test` mode is rarely used and it is the only mode, in which multiple lines are printed.
This separator only creates visual noise in the output, so I suggest to remove it. If you want to separate output from different ticks, then you can add, for example, a number of seconds passed from the test start and print it in the beggining of each line. In groups you can print it only on the first line, and for remaning lines add a spaces to align output